prov/efa/src/rdm/efa_rdm_pke_cmd.c

```diff
@@ -187,11 +187,17 @@ int efa_rdm_pke_fill_data(struct efa_rdm_pke *pkt_entry,
 		ret = efa_rdm_pke_init_dc_medium_tagrtm(pkt_entry, ope, data_offset, data_size);
 		break;
 	case EFA_RDM_DC_LONGCTS_MSGRTM_PKT:
-		assert(data_offset == 0 && data_size == -1);
+		/* The data_offset will be non-zero when the DC long CTS RTM packet
+		 * is sent to continue a runting read transfer after the
+		 * receiver has run out of memory registrations */
+		assert((data_offset == 0 || ope->internal_flags & EFA_RDM_OPE_READ_NACK) && data_size == -1);
```

Comment:
The original assertions should've been split into 2. Simpler to see which condition fails. Let's just fix that here.

```suggestion
		assert(data_offset == 0 || ope->internal_flags & EFA_RDM_OPE_READ_NACK);
		assert(data_size == -1);
```